pkg/collector/corechecks/containers/containerd.go

```diff
@@ -0,0 +1,359 @@
+			}
+			output.Tags = append(output.Tags, tags...)
+		}
+		sender.Event(output)
+	}
+}
+
+func computeMetrics(sender aggregator.Sender, nk context.Context, cu cutil.ContainerdItf, userTags []string) {
+	containers, err := cu.Containers()
+	if err != nil {
+		log.Errorf(err.Error())
+		return
+	}
+
+	for _, ctn := range containers {
+		t, errTask := ctn.Task(nk, nil)
+		if errTask != nil {
+			log.Tracef("Could not retrieve metrics from task %s: %s", ctn.ID()[:12], errTask.Error())
+			continue
+		}
+
+		tags, err := collectTags(ctn, nk)
+		if err != nil {
+			log.Errorf("Could not collect tags for container %s: %s", ctn.ID()[:12], err)
+		}
+		tags = append(tags, userTags...)
+		// Tagger tags
+		taggerTags, err := tagger.Tag(ctn.ID(), false)
+		if err != nil {
+			log.Errorf(err.Error())
+			continue
+		}
+		tags = append(tags, taggerTags...)
+
+		metrics, err := convertTasktoMetrics(t, nk)
+		if err != nil {
+			log.Errorf("Could not process the metrics from %s: %v", ctn.ID(), err.Error())
+			continue
+		}
+
+		err = computeExtra(sender, ctn, nk, tags)
+		if err != nil {
+			log.Errorf("Could not process metadata related metrics for %s: %v", ctn.ID()[:12], err.Error())
+		}
+
+		if metrics.Memory.Size() > 0 {
+			computeMem(sender, metrics.Memory, tags)
+		}
+
+		if metrics.CPU.Throttling != nil && metrics.CPU.Usage != nil {
+			computeCPU(sender, metrics.CPU, tags)
+		}
+
+		if metrics.Blkio.Size() > 0 {
+			computeBlkio(sender, metrics.Blkio, tags)
+		}
+
+		if len(metrics.Hugetlb) > 0 {
+			computeHugetlb(sender, metrics.Hugetlb, tags)
+		}
+	}
+}
+
+func convertTasktoMetrics(task containerd.Task, nsCtx context.Context) (*cgroups.Metrics, error) {
+	metricTask, err := task.Metrics(nsCtx)
+	if err != nil {
+		return nil, err
+	}
+
+	metricAny, err := typeurl.UnmarshalAny(&types.Any{
+		TypeUrl: metricTask.Data.TypeUrl,
+		Value:   metricTask.Data.Value,
+	})
+	if err != nil {
+		log.Errorf(err.Error())
+		return nil, err
+	}
+	return metricAny.(*cgroups.Metrics), nil
+}
+
+func computeExtra(sender aggregator.Sender, ctn containerd.Container, nsCtx context.Context, tags []string) error {
+	img, err := ctn.Image(nsCtx)
+	if err != nil {
+		return err
+	}
+	size, err := img.Size(nsCtx)
+	if err != nil {
+		return err
+	}
+	sender.Gauge("containerd.image.size", float64(size), "", tags)
+	return nil
+}
+
+func collectTags(ctn containerd.Container, nsCtx context.Context) ([]string, error) {
+	tags := []string{}
+
+	// Container image
+	im, err := ctn.Image(nsCtx)
+	if err != nil {
+		return tags, err
+	}
+	imageName := fmt.Sprintf("image:%s", im.Name())
+	tags = append(tags, imageName)
+
+	// Container labels
+	labels, err := ctn.Labels(nsCtx)
+	if err != nil {
+		return tags, err
+	}
+	for k, v := range labels {
+		tag := fmt.Sprintf("%s:%s", k, v)
+		tags = append(tags, tag)
+	}
+
+	// Container meta
+	i, err := ctn.Info(nsCtx)
+	if err != nil {
+		return tags, err
+	}
+
+	runt := fmt.Sprintf("runtime:%s", i.Runtime.Name)
```

Comment:
How would one use this tag?

Reply:
It's mostly for parity w/ the CRI check, not a lot of value I reckon but I figured that'd avoid confusion.
https://github.com/DataDog/datadog-agent/blob/master/pkg/collector/corechecks/containers/cri.go#L105